### PR TITLE
Fix LocalChannel close sequence

### PR DESCRIPTION
**Pull request by Scottmitch**
Motivation:
LocalChannel attempts to close its peer socket when ever it is closed. However if the channels are on different EventLoops we may attempt to process events for the peer channel on the wrong EventLoop.

Modifications:
- Ensure the close process ensures we are on the correct thread before accessing data

Result:
More correct LocalChannel close code.

### CODE REVIEW

**Comment by Scottmitch** [comment]
this should may help with #775

**Comment by Scottmitch** [comment]
@normanmaurer - FYI

File: transport/src/main/java/io/netty/channel/local/LocalChannel.java [hunk 1]
```diff
@@ -26,11 +26,14 @@
 import io.netty.channel.EventLoop;
 import io.netty.channel.SingleThreadEventLoop;
 import io.netty.util.ReferenceCountUtil;
+import io.netty.util.ReferenceCounted;
```
Comment (netkins) [inline transport/src/main/java/io/netty/channel/local/LocalChannel.java]: MINOR Remove this unused import 'io.netty.util.ReferenceCounted'. [rule](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by normanmaurer** [comment]
@Scottmitch LGTM

**Comment by Scottmitch** [comment]
4.1 (eb4d317) 4.0 (39f37a6)

**Comment by Scottmitch** [comment]
@normanmaurer - realized there was an issue after I pulled this in ... see https://github.com/netty/netty/pull/6053

**Pull request closed without merging**